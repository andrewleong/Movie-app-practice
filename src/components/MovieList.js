import React from 'react';

// Component imports
import MovieItem from './MovieItem';

const MovieList = ({ movieList=[] }) => {

    const generateList = (movies) => {
        if(movies.length){
            return movies.map( (item) => {
                return (
                    <div key={item.id}>
                        <MovieItem />
                    </div>
                )
            })
        }
    }

    return (
        <div className="col-xs-6 col-md-12">
            <div className="row">
                {generateList(movieList)}
            </div>
        </div>
    )
};

export default MovieList;
