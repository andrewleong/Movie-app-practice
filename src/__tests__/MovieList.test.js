import React from 'react';
import { shallow, mount } from 'enzyme';
import MovieList from '../components/MovieList';


describe('(Component) MovieList', () => {
    const wrapper = shallow(
        <MovieList
        />
    );

    const movies = [
        {
            id: 1,
            title: "Movie 1"
        },
        {
            id: 2,
            title: "Movie 2"
        }
    ]

    it('renders without crashing', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('accepts movieList props', () => {
        const wrapper = mount(
            <MovieList
                movieList={movies}
            />
        );
        expect(wrapper.props().movieList).toEqual(movies);
    });

    it('renders movieList correctly', () => {
        const wrapper = mount(
            <MovieList
                movieList={movies}
            />
        );
        expect(wrapper.props()).toMatchSnapshot();
    });

});
