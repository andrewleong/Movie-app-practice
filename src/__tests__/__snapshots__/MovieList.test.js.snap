// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) MovieList renders movieList correctly 1`] = `
Object {
  "movieList": Array [
    Object {
      "id": 1,
      "title": "Movie 1",
    },
    Object {
      "id": 2,
      "title": "Movie 2",
    },
  ],
}
`;
